sql/core/src/test/java/test/org/apache/spark/sql/JavaSparkSessionSuite.java

```diff
@@ -54,4 +55,19 @@ public void config() {
       Assert.assertEquals(spark.conf().get(e.getKey()), e.getValue().toString());
     }
   }
+
+  @Test
+  public void sqlParameters() {
+    spark = SparkSession.builder().master("local[*]").appName("testing").getOrCreate();
+    Map params = new HashMap();
+    params.put("_i1", "INTERVAL '1-1' YEAR TO MONTH");
+    params.put("p2", "'abc'");
```

Comment:
Could you add a test case where the string literal contains a double quote?

Reply:
added